src/framework/theme/components/popover/helpers/trigger.helper.ts

```diff
@@ -112,8 +117,8 @@ export class NbTriggerHelper {
    *
    * @return {NbPopoverTrigger} open and close events streams.
    * */
-  static createTrigger(host: HTMLElement, getContainer: Function, mode: NbPopoverMode): NbPopoverTrigger {
+  createTrigger(host: HTMLElement, getContainer: Function, mode: NbPopoverMode): NbPopoverTrigger {
     const createTrigger = NB_TRIGGERS[mode];
-    return createTrigger.call(NB_TRIGGERS, host, getContainer);
+    return createTrigger.call(NB_TRIGGERS, host, getContainer, this.document);
```

Comment:
Don't you think we shouldn't pass document for each trigger? Maybe, it would be better to move triggers in the NbTriggerService, to have shared context and access document through **this**?